### PR TITLE
Refactoring: Prevent js errors from parsing unsupported pages and remove try-catch blocks

### LINKED ISSUE #180: [Scalable Capital] JS-Fehler beim Auslesen einer Dividenden-PDF
<h2>Description</h2>
<blockquote>
<p>Ich @carstenhag hatte jetzt zum ersten Mal eine Scalable-Capital PDF importiert. Da die Seite komische Faxen gemacht hatte, hab ich die JS-Console geöffnet. Es wird ein Fehler beim Import geworfen. Korrekt importiert wird es aber komischerweise.</p>
</blockquote>
<pre><code>scalableCapital.js:227 Error while parsing page (scalable.capital) TypeError: Cannot read property 'substring' of undefined
    at findByStartingTerm (scalableCapital.js:57)
    at findISIN (scalableCapital.js:84)
    at parsePage (scalableCapital.js:188)
    at Module.parsePages (scalableCapital.js:225)
    at parseActivitiesFromPages (index.js:59)
    at _callee2$ (index.js:217)
    at tryCatch (pdf.worker.js:755)
    at Generator.invoke [as _invoke] (pdf.worker.js:930)
    at Generator.prototype.&lt;computed&gt; [as next] (pdf.worker.js:798)
    at asyncGeneratorStep (index.js:22) 
</code></pre>
<pre><code class="language-json">{
  &quot;broker&quot;: &quot;scalablecapital&quot;,
  &quot;type&quot;: &quot;Dividend&quot;,
  &quot;date&quot;: &quot;2020-10-07&quot;,
  &quot;isin&quot;: &quot;IE00B3RBWM25&quot;,
  &quot;company&quot;: &quot;Vanguard FTSE All-World U.ETF&quot;,
  &quot;shares&quot;: 0.635,
  &quot;price&quot;: 0.38546743314510523,
  &quot;amount&quot;: 0.25,
  &quot;fee&quot;: 0,
  &quot;tax&quot;: 0,
  &quot;filename&quot;: &quot;download (1).pdf_0&quot;,
  &quot;parsed&quot;: true,
  &quot;duplicate&quot;: true
}
</code></pre>
<h2>Document</h2>
<details><summary>download (1).pdf</summary><p>
<pre><code class="language-json">[
  [
    &quot;DCUP-050.041&quot;,
    &quot;Scalable Capital Vermögensverwaltung GmbH | Geschäftsführung: Erik Podzuweit, Florian Prucker | HRB 217778, Amtsgericht München |&quot;,
    &quot;USt-IdNr. DE300434774 | www.scalable.capital&quot;,
    &quot;Baader Bank Aktiengesellschaft Weihenstephaner Straße 4 85716 Unterschleißheim Deutschland T 00800 00 222 337* F +49 89 5150 2442&quot;,
    &quot;irardxv@gcawopicdl.rj www.baaderbank.de&quot;,
    &quot;* Kostenfreie Telefonnummer aus dem (inter-) nationalen Festnetz. Für Anrufe aus anderen Netzen können Gebühren anfallen.&quot;,
    &quot;Herausgeberin und verantwortlich für den Inhalt ist die Baader Bank Aktiengesellschaft.&quot;,
    &quot;Scalable Capital Vermögensverw.GmbH&quot;,
    &quot;Seitzstr. 8 e&quot;,
    &quot;80538 München&quot;,
    &quot;München&quot;,
    &quot;09.10.2020&quot;,
    &quot;Scalable Capital&quot;,
    &quot;Vermögensverwaltung GmbH&quot;,
    &quot;Seitzstraße 8e&quot;,
    &quot;80538 München&quot;,
    &quot;vfsmxbv@gzaddvab.hrdvszf&quot;,
    &quot;Dljloyi Yjiqghzt&quot;,
    &quot;Xkdsn-Nb.: 9504900    Vwxbnulkd: 9&quot;,
    &quot;Ehsuk-Bz.: 5639107133&quot;,
    &quot;Wefvzpdy-Mm.: 3268949&quot;,
    &quot;Saznbjvv-Qn.: 70477101&quot;,
    &quot;Fondsausschüttung&quot;,
    &quot;KOPIE&quot;,
    &quot;STK 0,635&quot;,
    &quot;Ausschüttung&quot;,
    &quot;Ex-Tag: 24.09.2020&quot;,
    &quot;ISIN: IE00B3RBWM25&quot;,
    &quot;WKN: A1JX52&quot;,
    &quot;Nominale&quot;,
    &quot;USD&quot;,
    &quot;0,454628 p.STK&quot;,
    &quot;Vanguard FTSE All-World U.ETF&quot;,
    &quot;Registered Shares USD Dis.oN&quot;,
    &quot;Zahlungszeitraum:&quot;,
    &quot;01.07.2019 - 30.06.2020&quot;,
    &quot;Zahltag:&quot;,
    &quot;07.10.2020&quot;,
    &quot;Umrechnungskurs:&quot;,
    &quot;1,17942&quot;,
    &quot;EUR/USD&quot;,
    &quot;Bruttobetrag&quot;,
    &quot;USD&quot;,
    &quot;0,29&quot;,
    &quot;Bruttobetrag&quot;,
    &quot;EUR&quot;,
    &quot;0,25&quot;,
    &quot;0,25&quot;,
    &quot;EUR&quot;,
    &quot;Valuta: 07.10.2020&quot;,
    &quot;Zu Gunsten Konto 1332876000&quot;,
    &quot;Dieses Dokument wurde maschinell erstellt und wird nicht unterschrieben.&quot;,
    &quot;Einkünfte aus Kapitalvermögen im Sinne von § 20 EStG sind einkommensteuerpflichtig.&quot;,
    &quot;Bitte beachten Sie Ihre eventuelle Meldepflicht nach § 67 AWV.&quot;,
    &quot;Seite 1/2&quot;
  ],
  [
    &quot;DCUP-050.041&quot;,
    &quot;Scalable Capital Vermögensverwaltung GmbH | Geschäftsführung: Erik Podzuweit, Florian Prucker | HRB 217778, Amtsgericht München |&quot;,
    &quot;USt-IdNr. DE300434774 | www.scalable.capital&quot;,
    &quot;Baader Bank Aktiengesellschaft Weihenstephaner Straße 4 85716 Unterschleißheim Deutschland T 00800 00 222 337* F +49 89 5150 2442&quot;,
    &quot;mjqxktj@iewutsgjgt.aj www.baaderbank.de&quot;,
    &quot;* Kostenfreie Telefonnummer aus dem (inter-) nationalen Festnetz. Für Anrufe aus anderen Netzen können Gebühren anfallen.&quot;,
    &quot;Herausgeberin und verantwortlich für den Inhalt ist die Baader Bank Aktiengesellschaft.&quot;,
    &quot;Fondsausschüttung&quot;,
    &quot;Fortsetzung:&quot;,
    &quot;KOPIE&quot;,
    &quot;Darstellung der steuerlichen Berechnungsgrundlagen:&quot;,
    &quot;vorher&quot;,
    &quot;nachher&quot;,
    &quot;Merkposten / Steuertöpfe&quot;,
    &quot;Erhöhung/Reduktion&quot;,
    &quot;Ausschüttung netto Aktienfonds&quot;,
    &quot;Allgemeiner Verlustverrechnungstopf&quot;,
    &quot;EUR&quot;,
    &quot;0,00&quot;,
    &quot;EUR&quot;,
    &quot;0,01&quot;,
    &quot;-&quot;,
    &quot;0,01&quot;,
    &quot;EUR&quot;,
    &quot;Ausschüttung netto Aktienfonds&quot;,
    &quot;Freistellungsauftrag&quot;,
    &quot;EUR&quot;,
    &quot;99,84&quot;,
    &quot;EUR&quot;,
    &quot;0,16&quot;,
    &quot;-&quot;,
    &quot;100,00&quot;,
    &quot;EUR&quot;,
    &quot;Ausschüttung netto Aktienfonds&quot;,
    &quot;Ausnutzung FSA&quot;,
    &quot;EUR&quot;,
    &quot;0,16&quot;,
    &quot;-&quot;,
    &quot;EUR&quot;,
    &quot;0,16&quot;,
    &quot;-&quot;,
    &quot;0,00&quot;,
    &quot;EUR&quot;,
    &quot;Steuerpflicht brutto&quot;,
    &quot;Steuerpflicht netto&quot;,
    &quot;Bemessungsgrundlagen&quot;,
    &quot;Ausschüttung brutto Aktienfonds&quot;,
    &quot;0,25&quot;,
    &quot;0,25&quot;,
    &quot;EUR&quot;,
    &quot;EUR&quot;,
    &quot;Ausschüttung netto Aktienfonds&quot;,
    &quot;0,17&quot;,
    &quot;0,00&quot;,
    &quot;EUR&quot;,
    &quot;EUR&quot;,
    &quot;Kapitalertragsteuer auf allgemeine Gewinne&quot;,
    &quot;0,17&quot;,
    &quot;0,00&quot;,
    &quot;EUR&quot;,
    &quot;EUR&quot;,
    &quot;Dieses Dokument wurde maschinell erstellt und wird nicht unterschrieben.&quot;,
    &quot;Einkünfte aus Kapitalvermögen im Sinne von § 20 EStG sind einkommensteuerpflichtig.&quot;,
    &quot;Bitte beachten Sie Ihre eventuelle Meldepflicht nach § 67 AWV.&quot;,
    &quot;Seite 2/2&quot;
  ]
]

</code></pre>
</p></details>
<h2>Community profile</h2>
<p>https://community.tresor.one/u/6f6e177f</p>
<hr />
<p>Issue submitted with https://import.tresor.one</p>

### LINKED ISSUE #257: [smartbroker] Verkauf Snowflake geht nicht zu importieren
## Document
<details><summary>Verkauf_Aktie_SNOWFLAKE_INC__A_DL-,0001_vom_20_11_2020.pdf</summary><p>

```json
[
  [
    "Frau",
    "Alessia Mayr",
    "Tegernseer Landstr. 28",
    "81541 München",
    "Alessia Mayr",
    "BNP Paribas S.A. Niederlassung Deutschland > Standort: München > Landsberger Straße 300 > 80687 München > Sitz: Nürnberg HRB Nürnberg 31129 > USt.-Ident-Nr.: DE",
    "191528929 > Sitz der Hauptniederlassung der BNP Paribas S.A.: 16, boulevard des Italiens, 75009 Paris, Frankreich, Registergericht: R.C.S. Paris 662 042 449 > Président du",
    "Conseil d‘Administration: Jean Lemierre, Directeur Général: Jean-Laurent Bonnafé",
    "Depot-Nr.",
    "5112863005",
    "Abrechnungs-Nr.",
    "35756275 / 20.11.2020",
    "Depotinhaber",
    "3.19/ABREABHNHANDVFDI/GAAASAFG/002938/201120/210623",
    "München, 20.11.2020",
    "BELEGDRUCK=J",
    "ORIGINAL=1",
    "FAXVERSAND=N",
    "EMAILVERSAND=N",
    "DEPOTNUMMER=5112863005",
    "DEPOTUNTERBEZEICHNUNG=",
    "VERSANDARTENSCHLUESSEL=DRUCK",
    "ADRESSZEILE1=Frau",
    "ADRESSZEILE2=Alessia Mayr",
    "ADRESSZEILE3=Tegernseer Landstr. 28",
    "ADRESSZEILE4=81541 München",
    "ADRESSZEILE5=",
    "ADRESSZEILE6=",
    "BELEGNUMMER=2938",
    "SEITENNUMMER=1",
    "STEUERERSTATTUNG=N",
    "Wertpapierabrechnung",
    "Verkauf",
    "Kommissionsgeschäft",
    "Gattungsbezeichnung",
    "Snowflake Inc. Reg. Shares Cl.A DL-,0001",
    "ISIN",
    "US8334451098",
    "Nominal",
    "STK 3,000",
    "Kurs",
    "EUR 226,0000",
    "Es folgt Seite",
    "2",
    "Wir haben für Sie verkauft",
    "Handelstag",
    "20.11.2020",
    "Handelszeit",
    "18:12*",
    "Börse",
    "Sekunden-Handel Aktien L&S",
    "Verwahrart",
    "Girosammelverwahrung",
    "Kurswert",
    "EUR",
    "678,00",
    "Provision",
    "EUR",
    "1,00",
    "-",
    "Kapitalertragsteuer",
    "EUR",
    "10,00",
    "-",
    "Solidaritätszuschlag",
    "EUR",
    "0,55",
    "-",
    "Wert",
    "24.11.2020",
    "Konto-Nr.",
    "5112863005",
    "Betrag zu Ihren Gunsten",
    "EUR",
    "666,45",
    "Hinweise zur steuerlichen Verrechnung:",
    "vorher",
    "aktuell",
    "Veräußerungsgewinn Aktien",
    "EUR",
    "40,00",
    "Aktienverlusttopf",
    "EUR",
    "0,00",
    "0,00",
    "allgemeiner Verlusttopf",
    "EUR",
    "0,00",
    "0,00",
    "Quellensteuertopf",
    "EUR",
    "0,00",
    "0,00",
    "zu versteuern",
    "EUR",
    "40,00",
    "im laufenden Jahr einbehaltene Kapitalertragsteuer",
    "EUR",
    "86,99",
    "im laufenden Jahr einbehaltener Solidaritätszuschlag",
    "EUR",
    "4,78",
    "Kapitalertragsteuer, Solidaritätszuschlag und ggf. Kirchensteuer nach gemeldetem Kirchensteuersatz verrechnet mit dem",
    "Finanzamt Frankfurt/M. V-Höchst, Steuernummer 047/220/15370.",
    "Jahressteuerbescheinigung folgt",
    "Wir werden in Ihrem Depot wie angegeben buchen.",
    "* Die angegebene Zeit entspricht den Angaben der Handelspartner gemäß der lokalen Zeit in Deutschland"
  ],
  [
    "BNP Paribas S.A. Niederlassung Deutschland > Standort: München > Landsberger Straße 300 > 80687 München > Sitz: Nürnberg HRB Nürnberg 31129 > USt.-Ident-Nr.: DE",
    "191528929 > Sitz der Hauptniederlassung der BNP Paribas S.A.: 16, boulevard des Italiens, 75009 Paris, Frankreich, Registergericht: R.C.S. Paris 662 042 449 > Président du",
    "Conseil d‘Administration: Jean Lemierre, Directeur Général: Jean-Laurent Bonnafé",
    "Depot-Nr.",
    "Abrechnungs-Nr.",
    "Seite-Nr.",
    "5112863005",
    "35756275",
    "2",
    "3.19/ABREABHNHANDVFDI/GAAASAFG/002938/201120/210623",
    "BELEGDRUCK=J",
    "ORIGINAL=1",
    "FAXVERSAND=N",
    "EMAILVERSAND=N",
    "DEPOTNUMMER=5112863005",
    "DEPOTUNTERBEZEICHNUNG=",
    "VERSANDARTENSCHLUESSEL=DRUCK",
    "ADRESSZEILE1=Frau",
    "ADRESSZEILE2=Alessia Mayr",
    "ADRESSZEILE3=Tegernseer Landstr. 28",
    "ADRESSZEILE4=81541 München",
    "ADRESSZEILE5=",
    "ADRESSZEILE6=",
    "BELEGNUMMER=2938",
    "SEITENNUMMER=2",
    "STEUERERSTATTUNG=N",
    "Dieser Beleg wird maschinell erstellt und daher nicht unterschrieben.",
    "(Mitteleuropäische Zeit bzw. Mitteleuropäische Sommerzeit)."
  ]
]

```

</p></details>

## Community profile

https://app.tresor.one/profile#

---
Issue submitted with https://import.tresor.one

### PR DESCRIPTION
**Pull request by dredav**
- This is a minor refactoring to prevent js errors from parsing invalid pages for: smartbroker, baaderbank and onvista
- Renamed `canParsePage` to `canParseFirstPage`
- Remove all try-catch blocks from `parsePages` to get handled as status code `6`
- Fix #257
- Fix #180

### CODE REVIEW

File: src/apps/portfolioPerformance/index.js [hunk 1]
```diff
@@ -173,7 +173,7 @@ export const parsePages = content => {
     .flatMap(validate);
 
   return {
-    activities,
+    activities: activities.filter(activity => activity !== undefined),
```
Comment (SirGibihm) [inline src/apps/portfolioPerformance/index.js:176]: Could you explain, maybe in a comment why activities might be undefined? I am of the opinion that undefined should never be returned for an activity. If this is the case, the import should fail

File: src/brokers/1822direkt.js [hunk 2]
```diff
@@ -173,24 +173,16 @@ export const parsePages = contents => {
...
+    if (activity === undefined) {
+      continue;
     }
+
+    activities.push(activity);
   }
 
   return {
-    activities,
+    activities: activities.filter(activity => activity !== undefined),
```
Comment (SirGibihm) [inline src/brokers/1822direkt.js:185]: Same as above, why wouldn't you have the doucment fail? Ignoring some lines within a document seems to be worse than secretly not importing.

File: src/brokers/baaderBank.js [hunk 3]
```diff
@@ -228,16 +228,23 @@ const findTax = content => {
...
   (isBrokerGratisbroker(content) ||
     isBrokerScalableCapital(content) ||
-    isBrokerOskar(content)) &&
-  (isPageTypeBuy(content) ||
-    isPageTypeSell(content) ||
-    isPageTypeDividend(content));
+    isBrokerOskar(content));
 
 const parsePage = content => {
+  if (!canParsePage(content) || content.some(line => line === 'Fortsetzung:')) {
```
Comment (SirGibihm) [inline src/brokers/baaderBank.js:244]: What does the string `Fortsetzung:` indicate?

File: src/brokers/baaderBank.js [hunk 4]
```diff
@@ -309,19 +316,11 @@ export const parsePages = contents => {
...
-        exception,
-        content
-      );
-    }
+    activities.push(parsePage(content));
   }
 
   return {
-    activities,
+    activities: activities.filter(activity => activity !== undefined),
```
Comment (SirGibihm) [inline src/brokers/baaderBank.js:323]: See above

File: src/brokers/comdirect.js [hunk 5]
```diff
@@ -472,7 +472,7 @@ export const parsePages = contents => {
   const activities = [parseData(contents.flat())];
 
   return {
-    activities,
+    activities: activities.filter(activity => activity !== undefined),
     status: 0,
```
Comment (SirGibihm) [inline src/brokers/comdirect.js:476]: See above

File: src/brokers/commerzbank.js [hunk 6]
```diff
@@ -369,7 +369,7 @@ export const parsePages = contents => {
     activities = [parseSingleTransaction(contents[0])];
   }
   return {
-    activities,
+    activities: activities.filter(activity => activity !== undefined),
```
Comment (SirGibihm) [inline src/brokers/commerzbank.js:372]: Not going to repeat myself with this change from now on :D

**Review by SirGibihm — CHANGES_REQUESTED** [review]

**Comment by dredav** [comment]
@SirGibihm I changes the logic for unsupported pages and resolve the merge conflicts.

**Review by SirGibihm — APPROVED** [review]
Looking good to me, give me the go ahaead an I will merge.

**Pull request merged**